### PR TITLE
Custom Store: fix `load(options)` type declaration

### PR DESCRIPTION
**Pull request by jacksonrr3**

### CODE REVIEW

File: js/data/custom_store.d.ts [hunk 1]
```diff
@@ -38,10 +39,10 @@ export interface CustomStoreOptions<
     /**
      * @docid
      * @type_function_param1 options:LoadOptions
-     * @type_function_return Promise<any>|Array<any>
+     * @type_function_return Promise<Array>|Array
      * @public
      */
-    load: ((options: LoadOptions<TItem>) => PromiseLike<TItem> | Array<TItem>);
+    load: ((options: LoadOptions<TItem>) => DxPromise<Array<TItem>> | Array<TItem>);
```
Comment (vconst) [inline js/data/custom_store.d.ts]: It is possible to return promise with { data: Array<TItem>, totalCount: number, summary: Array} object here.
Also grouped data can be returned.

Comment (vconst) [reply js/data/custom_store.d.ts]: See detail examples for jquery and angular here - https://js.devexpress.com/Documentation/Guide/Data_Binding/Specify_a_Data_Source/Custom_Data_Sources/

File: js/data/custom_store.d.ts [hunk 2]
```diff
@@ -1,12 +1,18 @@
...
 /** @public */
 export type Options<
     TItem = any,
     TKey = any,
 > = CustomStoreOptions<TItem, TKey>;
 
+/** @public */
+export type GroupItem<
+    TItem = any,
+> = { key: string; items?: Array<TItem> | GroupItem; count?: number; summary?: Array<number> };
```
Comment (vconst) [inline js/data/custom_store.d.ts]: key can have different types. So, key type should be changed to any or unknown
`items?: Array<TItem> | GroupItem` should be replaced to `items: Array<TItem> | Array<GroupItem<TItem>> | null`
Summary values can be not number. For example if use max summaryType for date or string column.

File: js/data/custom_store.d.ts [hunk 3]
```diff
@@ -38,10 +44,18 @@ export interface CustomStoreOptions<
     /**
      * @docid
      * @type_function_param1 options:LoadOptions
-     * @type_function_return Promise<any>|Array<any>
+     * @type_function_return Promise<Array|object>|Array
      * @public
      */
-    load: ((options: LoadOptions<TItem>) => PromiseLike<TItem> | Array<TItem>);
+    load: ((options: LoadOptions<TItem>) =>
+      | DxPromise<Array<TItem>
```
Comment (vconst) [inline js/data/custom_store.d.ts]: DxPromise<Array<TItem> -> DxPromise<Array<TItem> | Array<GroupItem<TItem>>

File: js/data/custom_store.d.ts [hunk 4]
```diff
@@ -38,10 +44,18 @@ export interface CustomStoreOptions<
...
+          | {
+              data: Array<TItem> | Array<GroupItem>;
```
Comment (vconst) [inline js/data/custom_store.d.ts]: Array<GroupItem> -> Array<GroupItem<TItem>>

File: js/data/custom_store.d.ts [hunk 5]
```diff
@@ -38,10 +44,18 @@ export interface CustomStoreOptions<
...
+              totalCount?: number;
+              summary?: Array<number>;
```
Comment (vconst) [inline js/data/custom_store.d.ts]: number should replaced to unknown or any because summary values can have different types.

**Review by vconst — CHANGES_REQUESTED** [review]

File: js/data/custom_store.d.ts [hunk 6]
```diff
@@ -38,10 +44,21 @@ export interface CustomStoreOptions<
     /**
      * @docid
      * @type_function_param1 options:LoadOptions
-     * @type_function_return Promise<any>|Array<any>
+     * @type_function_return Promise<Array|object>|Array
```
Comment (IlyaKhD) [inline js/data/custom_store.d.ts]:
```suggestion
     * @type_function_return Promise<Array<any>|object>|Array<any>
```

File: ts/dx.all.d.ts [hunk 7]
```diff
@@ -1674,7 +1680,19 @@ declare module DevExpress.data {
...
      * [descr:CustomStoreOptions.load]
      */
-    load: (options: LoadOptions<TItem>) => PromiseLike<TItem> | Array<TItem>;
+    load: (options: LoadOptions<TItem>) =>
+      | DevExpress.core.utils.DxPromise<
+          | Array<TItem>
+          | Array<DevExpress.data.CustomStore.GroupItem>
+          | {
+              data: Array<TItem> | Array<DevExpress.data.CustomStore.GroupItem>;
+              totalCount?: any;
```
Comment (vconst) [inline ts/dx.all.d.ts:1689]: totalCount type is number

**Review by vconst — APPROVED** [review]

**Pull request merged**